src/components/organisms/AdminVenueView/components/EventsView/EventsView.tsx

```diff
@@ -48,25 +56,50 @@ export const EventsView: React.FC<EventsViewProps> = ({ venueId, venue }) => {
 
   const renderedEvents = useMemo(
     () =>
-      events?.map((event) => {
-        return (
-          <TimingEvent
-            event={event}
+      events?.map((event) => (
+        <TimingEvent
+          event={event}
+          setShowCreateEventModal={setShowCreateEventModal}
+          setEditedEvent={setEditedEvent}
+          key={event.id}
+        />
+      )),
+    [events, setShowCreateEventModal, setEditedEvent]
+  );
+
+  const renderedSpaces = useMemo(() => {
+    const spaces = [...new Set(events?.map((event) => event.room))];
```

Comment:
Why the `Set`? `.map` returns a new array and should be the same result.
`const spaces = events?.map((event) => event.room)`

Reply:
Here I need a list of all portals inside the venue, and it should be unique for the render list

> `Set` is an object lets you store unique values of any type

Reply:
I get that but you are creating a Set with default indexes (which is 0,1,2,3 etc... just like a normal array) And then you spread the Set value into a new array. The code I suggested in my comment above should to the same trick

Reply:
discussed internally.

`const spaces = events?.map((event) => event.room)` is creating just an array of all spaces in the venue

> 0. "Posters"
> 1: ""
> 2: "testRoom"
> 3: "testRoom"
> 4: "testRoom"
> 5: "testRoom"
> 6: ""
> 7: "testRoom"
> 8: "testRoom"
> ...

and using Set makes it unique `const spaces = [...new Set(events?.map((event) => event.room))]`

> 0: "Posters"
> 1: ""
> 2: "testRoom"
> 3: "ohbmtest"